codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/RustCodegenPlugin.kt

```diff
@@ -34,22 +35,25 @@ class RustCodegenPlugin : SmithyBuildPlugin {
     override fun execute(context: PluginContext) {
         // Suppress extremely noisy logs about reserved words
         Logger.getLogger(ReservedWordSymbolProvider::class.java.name).level = Level.OFF
+        val codegenContext = context.toBuilder()
+            .model(ModelTransformer.create().flattenAndRemoveMixins(context.model))
+            .build()
```

Comment:
This works, but it should go into the `baselineTransform` function on `CodegenVisitor` since that's where all the rest of our model transforms are currently.